gef.py

```diff
@@ -9269,11 +9269,24 @@ def build_line(self, name: str, color: str, address_val: int, got_address: int)
         return line
 
     @only_if_gdb_running
-    def do_invoke(self, argv: List[str]) -> None:
+    @parse_arguments({"symbols": [""]}, {"--all": False})
+    def do_invoke(self, _: List[str], **kwargs: Any) -> None:
+        args : argparse.Namespace = kwargs["arguments"]
+        if args.all:
+            vmmap = gef.memory.maps
+            mapfiles = set(mapfile.path for mapfile in vmmap if
+                           pathlib.Path(mapfile.path).is_file() and
+                           mapfile.permission & Permission.EXECUTE)
+            for mapfile in mapfiles:
+                self.print_got_for(mapfile, args.symbols)
```

Comment:
I dont think this will work in this way when remoting, because `readelf` is checked locally whereas the mapped files will not point to local files. Either exclude the command from running when not local, or make sure the remote files are correctly tested (against the ones in the temporary namespace).

Reply:
I'll take another look at this one, tomorrow.

Reply:
> I'll take another look at this one, tomorrow.

No problem, thanks!
Also quick note, I've flagged your PR to be part of the next release.

Reply:
for remote does this normally work by downloading the binary offer, e.g. the file object points to the local copy?

Reply:
So... the list comprehension is looping over Section objects, and Section objects have a `.realpath` property that looks like it's intended to handle paths in the context of a remote session, so using that value seems like maybe it should be sufficient for this context.

But: the `got` command doesn't seem to work with remote sessions in the main branch.  I've checked out 29fb74efa29ec54296016d963724315e743689fa and run a `gef-remote` session to examine a process on a remote host.  It looks like all of the files have been copied to the local host, in `/tmp/tmp8alv77xz/`, but the `got` command fails:

```
(remote) gef➤  got
[!] Command 'got' failed to execute properly, reason: '/usr/libexec/postfix/master' not found/readable, most gef features will not work
```

I'm new to python extensions for gdb, so I'm actually not sure how to print the full exception/backtrace when something in gef fails, which makes it difficult for me to diagnose this further.

If everything else looks good, maybe this would be a good point to merge this PR, and then start a new PR to work on fixing remote use of the `got` command.  (Unless the failure of the pre-existing got command is the result of something I'm doing wrong.)